Source/MeshObjects/SMMeshObjects.f90

```diff
@@ -394,7 +394,7 @@ END FUNCTION IsOnOuterBox
 !
       PURE LOGICAL FUNCTION hasFixedPosition( node ) 
```

Comment:
Calling this function on line 174 of `Source/Mesh/LaplaceMeshSmoother.f90` throws a compilation error in `ifort`.
```
 ./Source/Mesh/LaplaceMeshSmoother.f90(174): error #8300: If a dummy argument is allocatable or a pointer, and the dummy or its associated actual argument is polymorphic, both dummy and actual must be polymorphic with the same declared type or both must be unlimited polymorphic.   [NODE]
               IF ( ASSOCIATED(node) .AND. .NOT.hasFixedPosition(node) )     THEN
-----------------------------------------------------------------^
compilation aborted for ./Source/Mesh/LaplaceMeshSmoother.f90 (code 1)
```

Do we need to declare the local variable in `Source/Mesh/LaplaceMeshSmoother.f90` to be
```fortran
TYPE(SMNode)              , POINTER  :: node     => NULL()
```
instead of `CLASS` to fix this error?